### PR TITLE
[server] Don't remove reports with the same report hash

### LINKED ISSUE #2952: Server removes reports with the same report hash
Steps to reproduce:
1. step: analyze and store the results to a run:
```cpp
// main.cpp
int main()
{
  sizeof(42);
  sizeof(42);
  sizeof(42);
}
```
2. step: update the file, reanalyze and store the results to the same run:
```cpp
// main.cpp
int main()
{
  sizeof(42);
}
```
In the second case the server will remove reports from the database which have the same report id. So in the database there will be actually one report.

### PR DESCRIPTION
**Pull request by csordasmarton**
Previously the server removed reports from the database which have the same
report hash on run update. With this patch we will not remove these reports
but we will set the detection status to Resolved.

### CODE REVIEW

File: web/tests/functional/db_cleanup/test_db_cleanup.py [hunk 1]
```diff
@@ -211,12 +211,13 @@ def test_garbage_file_collection(self):
         self._cc_client = env.setup_viewer_client(self.test_workspace)
         self.assertIsNotNone(self._cc_client)
 
-        self.assertEqual(len(files_in_report_before & files_in_report_after),
-                         0)
+        self.assertNotEqual(
```
Comment (gyorb) [inline web/tests/functional/db_cleanup/test_db_cleanup.py:214]: Why was the source file garbage collection behavior modified here?
If some source file was modified those source files should be removed and new source files should be added stored to that report?
Shouldn't be that tested here? However the test function name doesn't say much `test_garbage_file_collection`.
Should we check if the file ids are not the same before and after and the file ids which belong to the before part are not available anymore?

File: web/server/codechecker_server/api/report_server.py [hunk 2]
```diff
@@ -2660,29 +2669,22 @@ def get_analyzer_name(report):
...
-                    if checker in disabled_checkers:
-                        report.detection_status = 'off'
-                    elif checker_is_unavailable(checker):
-                        report.detection_status = 'unavailable'
-                    else:
-                        report.detection_status = 'resolved'
+        for reports in hash_map_reports.values():
+            for report in reports:
+                # We set the fix date of a report only if the report
+                # has not been fixed before.
```
Comment (gyorb) [inline web/server/codechecker_server/api/report_server.py:2675]: But how do we know if it was not fixed ever before? When it gets to the reopen state the fix date will be unset, do we collect somewhere else that the report was in the fixed state?

I just see in the code that the report is in currently in the fixed state and then it is skipped from the detection_status update.
Another question how can a report get into the reopened state? Is that updated somewhere else?

File: web/tests/functional/run_tag/test_run_tag.py [hunk 3]
```diff
@@ -172,13 +172,16 @@ def test_file_change(self):
         # Check the second file version.
         self._create_source_file(1, 'test_run_tag_update')
 
+        test_run_tags = self.__get_run_tag_counts(run_id, 100, 0)
+        run_tags_v0 = [t for t in test_run_tags if t.name == self.tags[0]]
```
Comment (gyorb) [inline web/tests/functional/run_tag/test_run_tag.py:176]: are there any other run tags mixed into the test data, that is why this run tag filtering needed?

**Review by gyorb — CHANGES_REQUESTED** [review]

**Comment by csordasmarton** [comment]
This will increment the database size drastically and we don't want to do this. For this reason we handled this problem in a different way. For more information see: #3102.

**Pull request closed without merging**